agent/runner/runner.go

```diff
@@ -124,40 +199,53 @@ func (r *Runner) ActionsResults() <-chan agentpb.AgentRequestPayload {
 
 // Stop stops running Action or Job.
 func (r *Runner) Stop(id string) {
-	r.rw.RLock()
-	defer r.rw.RUnlock()
+	r.cancelsM.RLock()
+	defer r.cancelsM.RUnlock()
 
-	// Job removes itself from rCancel map. So here we only invoke cancel.
-	if cancel, ok := r.rCancel[id]; ok {
+	// Job removes itself from cancels map. So here we only invoke cancel.
+	if cancel, ok := r.cancels[id]; ok {
 		cancel()
 	}
 }
 
 // IsRunning returns true if Action or Job with given ID still running.
 func (r *Runner) IsRunning(id string) bool {
```

Comment:
I just realized that message `JobStatusRequest` isn't sent by pmm-managed, so this function never will be called, so I'm not sure if it make sense to keep it here. seems like dead code